ShortestPath/ShortestPath/Network.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ShortestPath
+{
+    public class Network
+    {
+        public Dictionary<string,Arc> NetworkArcs = new Dictionary<string, Arc>();
+        public Dictionary<string, Node> NetworkNodes = new Dictionary<string, Node>();
```

Comment:
It seems that this Dictionary is not belonging to the Network class, but it is some state of the algorithm, am I wrong?

Reply:
No, these are Network fields that the other 2 ones were related to the algorithm status that I moved them to DijkstraAlgorithm class (ExtractedNodes Dictionary and heap that later convert it to interface)